Diagnostic/diagnostic.py

```diff
@@ -84,7 +84,25 @@ def init_distro_specific_actions():
     global g_dist_config, RunGetOutput
     dist = platform.dist()
     try:
-        g_dist_config = DistroSpecific.get_distro_actions(dist[0], dist[1], hutil.log)
+        if dist[0] != '':
+            name = dist[0]
+            version = dist[1]
+        else:
+            try:
+                # platform.dist() in python 2.7.15 does not recognize SLES/OpenSUSE.
+                with open("/etc/os-release") as fp:
+                    line = fp.readline()
```

Comment:
According to [the documentation](https://en.opensuse.org/SDB:Find_openSUSE_version), we should not be:

1. Assuming that the first two lines of /etc/os-release are NAME and VERSION. (As far as I can tell, no order is guaranteed)
2. Relying on NAME or VERSION as these [are meant to be human-readable fields](https://en.opensuse.org/SDB:Find_openSUSE_version). Instead, we should be looking at ID and VERSION_ID.

Reply:
Thx. Let me read the man page.